Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -47,6 +48,12 @@ public record DafnyDocument(
       TextDocumentItem.Text.Count(c => c == '\n') + 1
     );
 
+    // List of a few last touched method positions
+    public ImmutableList<Position> LastTouchedMethodPositions { get; set; } = new List<Position>() { }.ToImmutableList();
+
+    // Used to prioritize verification to one method and its dependencies
+    public Range? LastChange { get; init; } = null;
```

Comment:
Why use an `init;` property instead of making this part of the list of parameters in the primary constructor?

Reply:
Because this property cannot possibly be known at the time `new DafnyDocument` is usually created, so that mean we would need to add a default parameter to the record that is never assigned anything else.

That's a good indication for not using a record field, and use a Property instead, because it's "assigned" (like an immutable update in dafny) after a `new DafnyDocument` is returned, when some post-processing occurs.